lib/coffee/AccessTokenEncryptor.coffee

```diff
@@ -28,11 +35,31 @@ class AccessTokenEncryptor
 			crypted  = cipher.update(string, 'utf8', 'base64') + cipher.final('base64')
 			callback(null, @cipherLabel + ":" + salt.toString('hex') + ":" + crypted)
 
+	encryptJsonV2: (json, callback) ->
+		string = JSON.stringify(json)
+		crypto.randomBytes 32, (err, bytes) =>
+			return callback(err) if err
+			salt = bytes.slice(0, 16)
+			iv = bytes.slice(16, 32)
+
+			keyFn32 @cipherPassword, salt, 32, (err, key) =>
+				if err?
+					logger.err err:err, "error getting Fn key"
+					return callback(err)
+
+				cipher = crypto.createCipheriv(ALGORITHM, key, iv)
+				crypted = cipher.update(string, 'utf8', 'base64') + cipher.final('base64')
+
+				callback(null, "#{@cipherLabel}:#{salt.toString('hex')}:#{crypted}:#{iv.toString('hex')}")
+
 	decryptToJson: (encryptedJson, callback) ->
-		[label, salt, cipherText] = encryptedJson.split(':', 3)
+		[label, salt, cipherText, iv] = encryptedJson.split(':', 4)
+		if iv and iv.length > 0
+			return @decryptToJsonV2(encryptedJson, callback)
+
```

Comment:
I feel like this would be clearer if we used

```coffee
[label, salt, cipherText, iv] = encryptedJson.split(':', 4)
password = @settings.cipherPasswords[label]
return callback(new Error("invalid password")) if not password? or password.length < 16
if iv
  ... decrypTtoJsonV2(password, label, salt, cipherText, iv)
else
  ... decryptToJsonV1(password, label, salt, ciphertext)
```

to avoid duplication of splitting the string and checking the password.
No need to check iv.length here as it is a string, so any non-empty string will evaluate to `true`.

Reply:
>  No need to c No need to check iv.length here as it is a string, so any non-empty string will evaluate to true.heck iv.length here as it is a string, so any non-empty string will evaluate to true.

Showing my `c` roots again, aren't I?